assets/js/modules/analytics-4/datastore/report.js

```diff
@@ -157,6 +162,103 @@ const baseSelectors = {
 		return reports[ stringifyObject( options ) ];
 	},
 
+	/**
+	 * Gets a Page title to URL map for the given options.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state             Data store's state.
+	 * @param {Object} report            A report from the getReport selector containing pagePaths.
+	 * @param {Object} options           Options for generating the report.
+	 * @param {string} options.startDate Required, start date to query report data for as YYYY-mm-dd.
+	 * @param {string} options.endDate   Required, end date to query report data for as YYYY-mm-dd.
+	 * @return {(Object|undefined)} A map with url as the key and page title as the value. `undefined` if not loaded.
+	 */
+	getPageTitles: createRegistrySelector(
+		( select ) =>
+			( state, report, { startDate, endDate } = {} ) => {
+				if ( ! isPlainObject( report ) ) {
+					return;
+				}
+
+				const pagePaths = [];
+				const REQUEST_MULTIPLIER = 5;
```

Comment:
This `REQUEST_MULTIPLIER` value stems from https://github.com/google/site-kit-wp/issues/3070.